database/CoreDatabase.java

```diff
@@ -636,15 +661,27 @@ private void logSummary() {
         }
 
         public void committed(CoreTransaction.Data transaction) {
-            if (mayMiscount(transaction) && correctionRequired.compareAndSet(false, true)) {
+            handleDeferredSetUp();
+            if (mayMiscount(transaction) && state.compareAndSet(State.WAITING, State.CORRECTION_QUEUED)) {
                 submitCorrection();
+            }
+        }
 
+        private void handleDeferredSetUp() {
```

Comment:
we handle deferred intialisation, which is the code path that cluster relies on. It's not great to write this into the Core codebase, but this means cluster has to do minimal intervention